test/framework/style.py

```diff
@@ -60,6 +60,33 @@ def test_style_conformance(self):
 
         self.assertEqual(result, 0, "No code style errors (and/or warnings) found.")
 
+    def test_check_trailing_whitespace(self):
+        """Test for trailing whitespace check."""
+        lines = [
+            "name = 'foo'",  # no trailing whitespace
+            "version = '1.2.3'  ",  # trailing whitespace
+            "   ",  # blank line with whitespace included
+            '''description = """start of long description, ''',  # trailing whitespace, but allowed in description
+            ''' continuation of long description ''',  # trailing whitespace, but allowed in continued description
+            ''' end of long description"""''',
+            "moduleclass = 'tools'   ",  # trailing whitespace
+            '',
+        ]
+        line_numbers = range(1, len(lines) + 1)
+        test_cases = [
+            ({}, None),
+            ({}, (17, "W299 trailing whitespace")),
+            ({}, (0, "W293 blank line contains whitespace")),
+            ({}, None),
+            ({'eb_last_key': 'description'}, None),
+            ({'eb_last_key': 'description'}, None),
+            ({'eb_last_key': 'description'}, (21, "W299 trailing whitespace")),
```

Comment:
I should change this, I should be specifying the state manually, but just use `state = {}` before the loop and then pass in `state` without touching it in the test.